### ISSUE TITLE
2.9.0 Crashed after Skip login (Old Android)

### ISSUE THREAD

**Issue by LukasThyWalls** [twireapp/Twire#142]
I'm one of these weirdos with an old tablet with Android 4.1.1, and now the only option to see Twitch is Twire.

With 2.519, i can't login anymore but i can enter and see VODs and Lives if i search them, but i cannot see the thumbnails and images (supposedly fixed in 2.9.0 https://github.com/twireapp/Twire/commit/f79eb0a89bfb692cfa642f1bcf51ae2398459ec4).
Maybe it's possible in the near future to bookmark channels, or make a local follow system... but that's not the topic...

With 2.9.0, it crash when you enter the app. If i clean the app data, you can reach to skip the login, and then it crash.

Meanwhile, at least it works with 2.519...

**Comment by samfundev** [comment]
I'll need to look into reproducing this. If you could get me a log that would also help.

**Comment by LukasThyWalls** [comment]
```I/ActivityManager(  293): START {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.perflyst.twire/.activities.StartUpActivity bnds=[506,276][634,362] u=0} from pid 518
I/ActivityManager(  293): Start proc com.perflyst.twire for activity com.perflyst.twire/.activities.StartUpActivity: pid=2924 uid=10049 gids={3003, 1028}
V/ActivityManager(  293): ensureActivitiesVisible behind ActivityRecord{4243a378 com.perflyst.twire/.activities.StartUpActivity} configChanges=0x0
E/dalvikvm( 2924): Could not find class 'android.app.NotificationChannel', referenced from method com.perflyst.twire.TwireApplication.initNotificationChannels
W/dalvikvm( 2924): VFY: unable to resolve new-instance 44 (Landroid/app/NotificationChannel;) in Lcom/perflyst/twire/TwireApplication;
D/dalvikvm( 2924): DexOpt: unable to opt direct call 0x0124 at 0x1d in Lcom/perflyst/twire/TwireApplication;.initNotificationChannels
D/dalvikvm( 2924): DexOpt: unable to opt direct call 0x0124 at 0x2f in Lcom/perflyst/twire/TwireApplication;.initNotificationChannels
I/MultiDex( 2924): MultiDexExtractor(/data/app/com.perflyst.twire-1.apk, /data/data/com.perflyst.twire/code_cache/secondary-dexes)
I/MultiDex( 2924): Blocking on lock /data/data/com.perflyst.twire/code_cache/secondary-dexes/MultiDex.lock
I/MultiDex( 2924): /data/data/com.perflyst.twire/code_cache/secondary-dexes/MultiDex.lock locked
I/MultiDex( 2924): MultiDexExtractor.load(/data/app/com.perflyst.twire-1.apk, false, )
D/dalvikvm( 2924): Trying to load lib /data/data/com.perflyst.twire/lib/libconscrypt_jni.so 0x420379c8
D/dalvikvm( 2924): Added shared lib /data/data/com.perflyst.twire/lib/libconscrypt_jni.so 0x420379c8
I/ActivityManager(  293): START {cmp=com.perflyst.twire/.activities.setup.WelcomeActivity u=0} from pid 2924
V/ActivityManager(  293): ensureActivitiesVisible behind ActivityRecord{42482ff0 com.perflyst.twire/.activities.setup.WelcomeActivity} configChanges=0x480
I/ActivityManager(  293): Displayed com.perflyst.twire/.activities.setup.WelcomeActivity: +1s21ms (total +3s288ms)
I/ActivityManager(  293): START {flg=0x10000 cmp=com.perflyst.twire/.activities.setup.LoginActivity u=0} from pid 2924
V/ActivityManager(  293): ensureActivitiesVisible behind ActivityRecord{42376898 com.perflyst.twire/.activities.setup.LoginActivity} configChanges=0x0
I/ActivityManager(  293): Displayed com.perflyst.twire/.activities.setup.LoginActivity: +4s954ms
W/ActivityManager(  293): Activity idle timeout for ActivityRecord{42376898 com.perflyst.twire/.activities.setup.LoginActivity}
I/ActivityManager(  293): START {flg=0x10000 cmp=com.perflyst.twire/.activities.main.TopStreamsActivity u=0} from pid 2924
V/ActivityManager(  293): ensureActivitiesVisible behind ActivityRecord{42429770 com.perflyst.twire/.activities.main.TopStreamsActivity} configChanges=0x480
I/dalvikvm( 2924): Could not find method android.text.StaticLayout$Builder.obtain, referenced from method com.perflyst.twire.views.FontFitTextView.getStaticLayout
I/dalvikvm( 2924): Could not find method android.view.View.setTransitionName, referenced from method com.perflyst.twire.adapters.StreamsAdapter.handleElementOnClick
I/dalvikvm( 2924): Could not find method android.widget.PopupWindow.setElevation, referenced from method com.perflyst.twire.misc.TooltipWindow.showToolTip
E/AndroidRuntime( 2924): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.perflyst.twire/com.perflyst.twire.activities.main.TopStreamsActivity}: android.content.res.Resources$NotFoundException: File res/drawable/ic_add_circle.xml from drawable resource ID #0x7f0800a2
E/AndroidRuntime( 2924): 	at com.perflyst.twire.fragments.ChangelogDialogFragment.onCreateDialog(ChangelogDialogFragment.java:65)
W/ActivityManager(  293):   Force finishing activity com.perflyst.twire/.activities.main.TopStreamsActivity
W/ActivityManager(  293):   Force finishing activity com.perflyst.twire/.activities.setup.LoginActivity
W/ActivityManager(  293): Activity pause timeout for ActivityRecord{42429770 com.perflyst.twire/.activities.main.TopStreamsActivity}
V/ActivityManager(  293): ensureActivitiesVisible behind ActivityRecord{42482ff0 com.perflyst.twire/.activities.setup.WelcomeActivity} configChanges=0x480
W/System.err( 2924): 	at com.perflyst.twire.tasks.GetStreamsCountTask.doInBackground(GetStreamsCountTask.java:32)
W/System.err( 2924): 	at com.perflyst.twire.tasks.GetStreamsCountTask.doInBackground(GetStreamsCountTask.java:14)
I/ActivityManager(  293): Process com.perflyst.twire (pid 2924) has died.
W/ActivityManager(  293): Force removing ActivityRecord{42482ff0 com.perflyst.twire/.activities.setup.WelcomeActivity}: app died, no saved state
I/WindowState(  293): WIN DEATH: Window{425dbf70 com.perflyst.twire/com.perflyst.twire.activities.setup.LoginActivity paused=true}
I/WindowState(  293): WIN DEATH: Window{42597280 com.perflyst.twire/com.perflyst.twire.activities.setup.WelcomeActivity paused=false}
D/InstalledAppProviderSer( 2139): deleteAppFromDb com.perflyst.twire
D/InstalledAppProviderSer( 2139): insertAppIntoDb com.perflyst.twire
```

I used _aLogCat ROOT_ with filter "Twire". I hope it's useful.

**Comment by KandiPanda** [comment]
Android 4.1.2 on a Casio C811 here, Was trying out Twire.

2.9.0 crashes if I skip login. Downgraded to 2.519, skipping login works fine, lets me watch streams at least.

Either version, attempting to log in takes me to a page reading:
```
Webpage not available

The webpage at https://www.twitch.tv/login?client_id=(bunch of stuff here) might be temporarily down or it may have moved permanently to a new web address.
```

**Comment by KandiPanda** [comment]
2.9.1 also does this, same as 2.9.0 does.

**Comment by MaxCoveri** [comment]
> 2.9.1 also does this, same as 2.9.0 does.

Tried 2.9.2 on an Android TV 4.4.2, it also has this problem. Skipping login crashes the app, and trying to log in leaves the app stuck in the "sit tight, collecting your follows" screen.

**Comment by kevingnr** [comment]
@samfundev
Error Logs from Android Studio tested on an Android 4.2.2:

```
25775-25775/com.perflyst.twire.debug E/AndroidRuntime: FATAL EXCEPTION: main
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.perflyst.twire.debug/com.perflyst.twire.activities.main.TopStreamsActivity}: android.content.res.Resources$NotFoundException: File res/drawable/ic_bug_fix.xml from drawable resource ID #0x7f0800d1
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2343)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395)
        at android.app.ActivityThread.access$600(ActivityThread.java:162)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1364)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:194)
        at android.app.ActivityThread.main(ActivityThread.java:5392)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:525)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/ic_bug_fix.xml from drawable resource ID #0x7f0800d1
        at android.content.res.Resources.loadDrawable(Resources.java:1993)
        at android.content.res.Resources.getDrawable(Resources.java:673)
        at android.content.res.XResources.getDrawable(XResources.java:521)
        at androidx.core.content.ContextCompat.getDrawable(ContextCompat.java:457)
        at com.perflyst.twire.fragments.ChangelogDialogFragment.onCreateDialog(ChangelogDialogFragment.java:70)
        at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:419)
        at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1484)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
        at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447)
        at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2629)
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1170)
        at android.app.Activity.performStart(Activity.java:5132)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2316)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395) 
        at android.app.ActivityThread.access$600(ActivityThread.java:162) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1364) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:194) 
        at android.app.ActivityThread.main(ActivityThread.java:5392) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:525) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600) 
        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
        at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:881)
        at android.graphics.drawable.Drawable.createFromXml(Drawable.java:822)
        at android.content.res.Resources.loadDrawable(Resources.java:1990)
        at android.content.res.Resources.getDrawable(Resources.java:673) 
        at android.content.res.XResources.getDrawable(XResources.java:521) 
        at androidx.core.content.ContextCompat.getDrawable(ContextCompat.java:457) 
        at com.perflyst.twire.fragments.ChangelogDialogFragment.onCreateDialog(ChangelogDialogFragment.java:70) 
        at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:419) 
        at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1484) 
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320) 
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187) 
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356) 
        at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434) 
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497) 
        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447) 
        at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169) 
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992) 
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947) 
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849) 
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2629) 
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577) 
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247) 
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541) 
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210) 
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1170) 
        at android.app.Activity.performStart(Activity.java:5132) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2316) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395) 
        at android.app.ActivityThread.access$600(ActivityThread.java:162) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1364) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:194) 
        at android.app.ActivityThread.main(ActivityThread.java:5392) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:525) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600) 
        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132) 
        at dalvik.system.NativeStart.main(Native Method) 
25775-25836/com.perflyst.twire.debug E/dalvikvm: Could not find class 'javax.net.ssl.X509ExtendedTrustManager', referenced from method org.conscrypt.ConscryptEngineSocket.getDelegatingTrustManager
```

**Comment by KandiPanda** [comment]
Hmm... Looking at that error log and doing a little searching, could the problem be somehow related to these issues on conscrypt's github page?

https://github.com/google/conscrypt/issues/603

https://github.com/google/conscrypt/issues/848

**Comment by vapormusic** [comment]
Should be fixed here #196

**Comment by samfundev** [comment]
I'm going to mark this issue as closed now that #196 has been merged in.
